utils/fetch_remote.go

```diff
@@ -87,13 +86,9 @@ func ListRemotes() (map[string][]string, error) {
 			splitPath := strings.Split(u.Path, "/")
 			owner = splitPath[0]
 		} else if strings.HasPrefix(remoteURL, "https://") {
-			u, err := url.Parse(remoteURL)
-			if err != nil {
-				continue
-			}
-			splitPath := strings.Split(u.Path, "/")
-
-			owner = splitPath[0]
+			remoteRegexp := regexp.MustCompile(`.+//(.+)/(.+)/(.+)(\.git)?$`)
+			matched := remoteRegexp.FindStringSubmatch(remoteURL)
+			owner = matched[2]
```

Comment:
@burntcarrot Thanks for this. The `ListRemotes` function is critical for several commands in the CLI which require the repository details. Would be great if you can somehow add unit tests for the remotes extraction logic. Feel free to refactor the function and use dummy input of `git` commands in order to make it easy for writing tests.

Reply:
Added unit tests for the remote extraction logic. Had to refactor some code too. 😓